### PR TITLE
[JH] [MF] Add Item to Shopping List

### LINKED ISSUE #5: 5. As a user, I want to add a new item to my shopping list so I can start recording purchases
## Summary
Users need to be able to add items to their shopping lists. To do this, the `AddItem` view should present them with a form that allows them to enter two things:
1. The name of the item, and
2. An estimate of when they think they’ll need to buy the item again.

[The README file for the shopping list API](https://github.com/the-collab-lab/smart-shopping-list-next/blob/main/src/api/README.md) will be helpful as you work on this feature.

## Acceptance criteria
UI-related tasks:
- [x] The `ManageList` view displays a form that allows them to enter the name of the item and select how soon they anticipate needing to buy it again. There should be 3 choices for this:
	- “Soon”, corresponding to 7 days
	- “Kind of soon”, corresponding to 14 days
	- “Not soon”, corresponding to 30 days
- [x] The input that accepts the name of the item has a semantic `label` element associated with it
- [x] The user can submit this form with both the mouse and the `Enter` key  
- [x] When the user submits the form, they see a message indicating that the item either was or was not saved to the database.

Data-related tasks:
- [x] The `console.log` in the `addItem` function in `src/api/firebase.js`  is replaced with a function that adds the new document to the Firestore database. That function will be imported from the `firebase/firestore` module.
- [x]  The user’s soon/not soon/kind of soon choice is used to calculate `nextPurchasedDate`

## Notes
- When this feature is implemented correctly, new items will *automatically* show up in the list view because `streamListItem` gets new data from Firestore every time there’s a change.
- You will know your `<label>` is correctly implemented if you click on the label and keyboard focus moves to the related input. Refer to [the MDN docs on `<label>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label) for more info.
- Users will be able to submit the form with the `Enter` key if you listen for the right JavaScript event on the right element! If you're adding an `onKeyDown` listener anywhere, you might be overcomplicating things.
- The add item form has the potential to be a lot of scope. Stick to the Acceptance Criteria.

### PR DESCRIPTION
**Pull request by hsiangj**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description

Added form in ManageList.jsx for users to add items to their shopping lists, where users are able to provide new item name and when they might need to buy the item again. New items will automatically show on List page.

When addressing the addItem Firebase functionality in the Acceptance Criteria, we learned the data model no longer required as many nested collections/documents. The solution we implemented allowed us to remove the double map in List.jsx. 

## Related Issue

Closes #5 

## Acceptance Criteria

UI-related tasks:
- [x] The `ManageList` view displays a form that allows them to enter the name of the item and select how soon they anticipate needing to buy it again. There should be 3 choices for this:
	- “Soon”, corresponding to 7 days
	- “Kind of soon”, corresponding to 14 days
	- “Not soon”, corresponding to 30 days
- [x] The input that accepts the name of the item has a semantic `label` element associated with it
- [x] The user can submit this form with both the mouse and the `Enter` key  
- [x] When the user submits the form, they see a message indicating that the item either was or was not saved to the database.

Data-related tasks:
- [x] The `console.log` in the `addItem` function in `src/api/firebase.js`  is replaced with a function that adds the new document to the Firestore database. That function will be imported from the `firebase/firestore` module.
- [x]  The user’s soon/not soon/kind of soon choice is used to calculate `nextPurchasedDate`

## Type of Changes

Use one or more labels to help your team understand the nature of the change(s) you’re proposing. E.g., `bug fix` or `enhancement` are common ones.

### Before
Screenshot 2024-02-13 at 5 58 06 PM

#### Add Item Form
Screenshot 2024-02-13 at 5 57 09 PM

#### Success message when item is added to database
Screenshot 2024-02-13 at 6 01 41 PM

## Testing Steps / QA Criteria
Using either the pulled version or test deploy: on Home page, click desired list. Navigate to 'Manage-List' to add item. After adding item, navigate to 'List' to view new item.

### CODE REVIEW

**Comment by piecanoe** [comment]
Amalya and I reviewed together and this looks good! It passes all the QA tests.

**Review by piecanoe — APPROVED** [review]

**Comment by louieknolle** [comment]
Looks good and works as expected. Form is navigable with keyboard and mouse and items are added and visible on the List page. 👍🏻👍🏻

**Review by louieknolle — APPROVED** [review]

File: src/views/List.jsx [hunk 1]
```diff
@@ -8,9 +8,10 @@ export function List({ data }) {
 			</p>
 			<ul>
 				{data && data.length > 0 ? (
-					data.map((item) =>
-						item.items.map((name, id) => <ListItem key={id} name={name} />),
-					)
+					data.map((item, id) => (
+						// Removed double map to reflect current Firestore data model
```
Comment (rmorabia) [inline src/views/List.jsx]: This would be great as a GitHub comment, not necessarily a code comment!

**Review by rmorabia — APPROVED** [review]
Great job! I added one nitpick. If you don't want nitpicks, just let me know and I can tone it down. :)

File: src/views/List.jsx (hunk 1 above)
Comment (hsiangj) [reply src/views/List.jsx]: This is helpful, will remove thanks!

**Pull request merged**